azurelinuxagent/ga/exthandlers.py

```diff
@@ -486,33 +491,44 @@ def handle_enable(self, ext_handler_i):
             ext_handler_i.update_settings()
 
         ext_handler_i.enable()
+        # Remove the UNINSTALL_FAILED from env variables list after calling Install and Enable
+        ext_handler_i.remove_key_from_handler_env(UNINSTALL_FAILED)
 
     @staticmethod
     def _update_extension_handler(old_ext_handler_i, ext_handler_i):
 
-        def execute_old_handler_command(func):
+        def execute_old_handler_command(op, func):
             """
             Created a common wrapper to execute all commands that need to be executed from the old handler
             so that it can have a common exception handling mechanism
+            :param op: The current running operation
             :param func: The command to be executed on the old handler
             """
+            continue_on_update_failure = False
             try:
+                continue_on_update_failure = ext_handler_i.load_manifest().is_continue_on_update_failure()
                 func()
             except ExtensionError as e:
                 # Reporting the event with the old handler and raising a new ExtensionUpdateError to set the
                 # handler status on the new version
-                msg = ustr(e)
+                msg = "%s; ContinueOnUpdate: %s" % (ustr(e), continue_on_update_failure)
```

Comment:
Logging the status of ContinueOnUpdate flag incase of errors

Reply:
We should also log which operation failed (`op`) in the message

Reply:
The ExtensionError exception message `e` would already have the specifics of weather disable or uninstall failed. I feel logging the `op` again would just be redundant

Reply:
But the call to `report_event` doesn't have the operation, which means we couldn't tell from our telemetry if disable or uninstall failed.

Reply:
Why do we initialize continue_on_update_failure within the try except? thanks

Reply:
@pgombar we set the operation on every call to disable/enable/install/etc (like `self.set_operation(WALAEventOperation.Disable)`). If there's an error thrown then the report event just uses that same operation to report the event `op=self.operation,`. Thanks for pointing this out though, I'll reuse the same `op=self.operation` in my function instead of passing another op parameter.

Reply:
@narrieta I added it to the try catch because the load_manifest might throw an ExtensionError which we should catch and report as an ExtensionUpdateError (since it is being caught during the update sequence)

```python
continue_on_update_failure = ext_handler_i.load_manifest().is_continue_on_update_failure()


    def load_manifest(self):
        man_file = self.get_manifest_file()
        try:
            data = json.loads(fileutil.read_file(man_file))
        except (IOError, OSError) as e:
            raise ExtensionError('Failed to load manifest file ({0}): {1}'.format(man_file, e.strerror),
                                 code=ExtensionErrorCodes.PluginHandlerManifestNotFound)
        except ValueError:
            raise ExtensionError('Malformed manifest file ({0}).'.format(man_file),
                                 code=ExtensionErrorCodes.PluginHandlerManifestDeserializationError)

        return HandlerManifest(data[0])
```